src/com/example/api/ConnectionTask.java

```
@@ -0,0 +1,89 @@
+package com.example.api;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: sizhao
+ * Date: 6/29/13
+ * Time: 7:46 PM
+ * To change this template use File | Settings | File Templates.
+ */
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: rthai
+ * Date: 6/26/13
+ * Time: 3:49 PM
+ * To change this template use File | Settings | File Templates.
+ */
+public class ConnectionTask extends AsyncTask<URL, Void, String> {
+    protected String doInBackground(URL... urls) {
+        InputStream is = null;
+        String contentAsString = "";
+
+        try {
+            URL url = urls[0];
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setReadTimeout(10000 /* milliseconds */);
+            conn.setConnectTimeout(15000 /* milliseconds */);
+            conn.setRequestMethod("GET");
+            conn.setDoInput(true);
+            // Starts the query
+            conn.connect();
+            int response = conn.getResponseCode();
+            Log.d("debug_tag", "The response is: " + response);
+            is = conn.getInputStream();
+
+            // Convert the InputStream into a string
+            contentAsString = readIt(is);
+
+            // Makes sure that the InputStream is closed after the app is
+            // finished using it.
+        } catch (ProtocolException e) {
+            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
```

You never want to have e.printStackTrace in production code.  It almost always means that you're not handling an exception correctly.  Usually the answer is to `throw new RuntimeException(e)` so you can propagate the error.  If you really want to swallow the error without propagating it, you can use Log.e() and add a comment that says `// intentionally ignored` or something to inform people that you are deliberately swallowing the error.